tools/merge_archives.ml

```diff
@@ -14,8 +14,22 @@
 
 module String_set = Set.Make (String)
 
-let merge_cma ~target:_ ~archives:_ =
-  failwith "not yet implemented"
+let merge_cma ~target ~archives =
+  (* This is rather tightly tied to the internals of [Bytelibrarian]. *)
+  Clflags.link_everything := false;
+  Clflags.custom_runtime := false;
+  Clflags.no_auto_link := false;
+  Clflags.ccobjs := [];
+  Clflags.all_ccopts := [];
+  Clflags.dllibs := [];
+  List.iter
+    (fun archive -> Load_path.add_dir (Filename.dirname archive))
+    archives;
+  try Bytelibrarian.create_archive archives target
```

Comment:
Note: I think the reason why we get the same semantics
as for `.cmxa` files is because warning 31 is not only on
by default, but also in the warning-as-error set.
We probably need to call `Warnings.{reset,check}_fatal`
to fail properly, though.

Reply:
If warning 31 is now emitted the program fails (I checked by duplicating an entry in the `.cma` list in the dune file).